plugins/cachix/auth_token.go

```diff
@@ -0,0 +1,42 @@
+package cachix
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func AuthToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.AuthToken,
+		DocsURL:       sdk.URL("https://docs.cachix.org/getting-started#authenticating"),
+		ManagementURL: sdk.URL("https://app.cachix.org/personal-auth-tokens"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to Cachix.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					// Cache auth token consists of 149 chars whereas personal auth token consists of
+					// 145 chars.
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
```

Comment:
The tokens seem to be JWT's (see my other comment). That means that we can be just a little more specific about the composition:

```suggestion
					// The tokens are JWT's.
					Charset: schema.Charset{
						Uppercase: true,
						Lowercase: true,
						Digits:    true,
						Specific:  []rune{'.', '-', '_'},
					},
```